ParkingFinder/handlers/parking_space.py

```diff
@@ -122,4 +126,80 @@ def post(self, user_id):
             })
 
 
+class ParkingSpaceRealTimeHandler(BaseHandler):
+
+    @coroutine
+    def get(self, user_id):
+
+        try:
+            access_token = self.get_argument('access_token', default=None)
+            # TODO: token validation
+            assert user_id
+            
+            realtime = yield ParkingSpaceService.initialize_waiting_user_location(user_id)
+
+            realtime_res = RealTimeMapper.to_record(realtime)
+            self.set_status(httplib.OK)
+            self.write({
+                'realtime': realtime_res
+            })
+
+        except AssertionError:
+            self.set_status(httplib.BAD_REQUEST)
+            self.write({
+                'error': "BAD REQUEST Invalid User Id"
+            })
+        except MissingArgumentError:
+            self.set_status(httplib.BAD_REQUEST)
+            self.write({
+                'error': "BAD REQUEST Location Information Is Incomplete"
+            })
+        except NotFound:
+            self.set_status(httplib.NOT_FOUND)
+            self.write({
+                'error': "user doesn't match any other users"
+            })
+
+    @coroutine
+    def post(self, user_id):
+
+        try:
+            access_token = self.get_argument('access_token', default=None)
+            # TODO: token validation
+            assert user_id
+
+            request_body = json.loads(self.request.body)
+            _realtime = request_body.get('realtime', False)
+
```

Comment:
You can simply use Realtime(_realtime) to create the entity. And use validate method if needed. However, as I mentioned below, the request should only contains the location who sent the request. You can use waiting user as key, even though I still think this is confusing others who read you code without explanation.